src/main/java/io/supertokens/authRecipe/AuthRecipe.java

```diff
@@ -54,4 +54,11 @@ public static UserPaginationContainer getUsers(Main main, Integer limit, String
         System.arraycopy(users, 0, resultUsers, 0, maxLoop);
         return new UserPaginationContainer(resultUsers, nextPaginationToken);
     }
+
+    public static void deleteUser(Main main, String userId) throws StorageQueryException {
+        StorageLayer.getEmailPasswordStorage(main).deleteUser(userId);
+        StorageLayer.getThirdPartyStorage(main).deleteThirdPartyUser(userId);
+        StorageLayer.getEmailVerificationStorage(main).deleteUserInfo(userId);
+        StorageLayer.getSessionStorage(main).deleteSessionsOfUser(userId);
```

Comment:
We should delete users at the very end. Therefore, we should remove sessions, email verification info, reset password info before.